exporters/dsexporter/dsexporter.go

```diff
@@ -1,39 +1,108 @@
 package main
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
 	"net/http"
+	"os"
+	"strings"
+	"errors"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
 )
 
+const service = "grafana"
+const check = "prometheus-appstudio-ds"
+var allDataSources = GetDataSources
+
 type CustomCollector struct {
-	requestCounter prometheus.Counter
+	konfluxUp *prometheus.GaugeVec
 }
 
 // Creating a new instance of CustomCollector.
 func NewCustomCollector() *CustomCollector {
 	return &CustomCollector{
-		requestCounter: prometheus.NewCounter(prometheus.CounterOpts{
-			Name: "request_count",
-			Help: "Number of requests handled by the handler",
-		}),
+		konfluxUp: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "konflux_up",
+			Help: "Availability of the Konflux default grafana datasource",
+		},
+		[]string{"service", "check"}),
 	}
 }
 
 // Describe method sends descriptions of the metrics to Prometheus.
 // When Prometheus scrapes the /metrics endpoint of the exporter,
 // it first calls the Describe method to get a description of all the metrics.
 func (e *CustomCollector) Describe(ch chan<- *prometheus.Desc) {
-	e.requestCounter.Describe(ch)
+	e.konfluxUp.Describe(ch)
 }
 
 // Collect method sends the current values of the metrics to Prometheus.
 // After Prometheus understands what metrics are available (using the `Describe` method),
 // it then calls the `Collect` method to actually get the values of those metrics.
 func (e *CustomCollector) Collect(ch chan<- prometheus.Metric) {
-	e.requestCounter.Collect(ch)
+	var availability float64
+	clientset := kubernetes.NewForConfigOrDie(config.GetConfigOrDie())
+
+	if grafanaRes, err := allDataSources(GetGrafanaResource(clientset)); err == nil {
+		availability = GetDataSourceAvailability(grafanaRes, check)
+	}
+
+	e.konfluxUp.WithLabelValues(service, check).Set(availability)
+	e.konfluxUp.Collect(ch)
+}
+
+// get the grafana resource as a map
+func GetGrafanaResource(clientset *kubernetes.Clientset) map[string]interface{} {
+	data, err := clientset.RESTClient().
+		Get().
+		AbsPath("/apis/grafana.integreatly.org/v1beta1").
+		Namespace("appstudio-grafana").
+		Resource("grafanas").
+		Name("grafana-oauth").
+		DoRaw(context.TODO())
+	var grafanaResource map[string]interface{}
+	err = json.Unmarshal(data, &grafanaResource)
+	if err != nil {
+		fmt.Printf("Error getting resource: %v\n", err)
+		os.Exit(1)
```

Comment:
@yftacherzog @avi-biton @amisstea Here as well should I modify to return error message and then return metric value as 0 instead of exiting?

Reply:
Yes. I think that's the way to go.

Reply:
Updated it.